api/tests/app/test_app.py

```diff
@@ -0,0 +1,162 @@
+        assert response.status_code == 200
+        assert response.json() == expected_response
+
+    # pytest -v api/tests/app/test_app.py::TestClassUpdateAppConfig::test_try_change_app_config_wrong_type_variable_all
+    @pytest.mark.parametrize("key_value_dict, correct_type", [
+        ({"has_been_configured": "Here_must_be_a_bool_variable"}, "bool"),
+        ({"has_been_configured": 40}, "bool"),
+        ({"resolution": False}, "string"),
+        ({"resolution": 40}, "string"),
+        ({"encoder": False}, "string"),
+        ({"encoder": 40}, "string"),
+        ({"max_processes": "Here_must_be_an_integer_variable"}, "integer"),
+        ({"max_processes": False}, "integer"),
+        ({"dashboardurl": 40}, "string"),
+        ({"dashboardurl": False}, "string"),
+        ({"screenshots_directory": 40}, "string"),
+        ({"screenshots_directory": False}, "string"),
+        ({"slack_channel": 40}, "string"),
+        ({"slack_channel": False}, "string"),
+        ({"occupancy_alerts_min_interval": False}, "integer"),
+        ({"occupancy_alerts_min_interval": "Here_must_be_an_integer_variable"}, "integer"),
+        ({"max_thread_restarts": False}, "integer"),
+        ({"max_thread_restarts": "Here_must_be_an_integer_variable"}, "integer"),
+        ({"global_reporting_emails": 40}, "string"),
+        ({"global_reporting_emails": False}, "string"),
+        ({"global_report_time": 40}, "string"),
+        ({"global_report_time": False}, "string"),
+        ({"daily_global_report": "Here_must_be_a_bool_variable"}, "bool"),
+        ({"daily_global_report": 40}, "bool"),
+        ({"weekly_global_report": "Here_must_be_a_bool_variable"}, "bool"),
+        ({"weekly_global_report": 40}, "bool"),
+        ({"heatmap_resolution": 40}, "string"),
+        ({"heatmap_resolution": False}, "string"),
+    ])
+    def test_try_change_app_config_wrong_type_variable_all(self, config_rollback, key_value_dict,
+                                                           correct_type):
+        """
+        It is important to mention, the fact that if pydantic expects a string, it will try to
+        cast what was sent, so if we send a bool or an integer to a string field, pydantic will work:
+          Example:
+              a = bool or integer
+              str(a)
+          Pydantic will accept it and PUT request will work.
+
+        Idem when we send a bool, and an integer is expected.
+            int(False) = 0
+            int(True) = 1
+        """
+        client, config_sample_path = config_rollback
+
+        body = create_app_config(key_value_dict)
+
+        response = client.put('/app', json=body)
+
+        if correct_type == "string":
+            assert response.status_code == 200
+            expected_response = expected_response_function(config_sample_path)
+            assert response.json() == expected_response
+        elif correct_type == "integer":
```

Comment:
Is missing the `correct_type == "integer"` case?

Reply:
I think you meant the "bool" case, I made it explicit now.